fw/sync_socket.h

```diff
@@ -32,6 +32,7 @@
 typedef struct ss_proto_t {
 	const struct ss_hooks	*hooks;
 	int			type;
+	int			option;
```

Comment:
> Can we override `type` when we know protocol? By default(no alpn) we will use HTTP2 in case when HTTP2 is specified to socket. Also, we always must prefer HTTP2 over HTTP1.

I rewrite code without "option"